### PR TITLE
Standardize toml format with taplo

### PR DESCRIPTION
**Pull request by ameknite**
# Objective

- Standardize fmt for toml files

## Solution

- Add [taplo](https://taplo.tamasfe.dev/) to CI (check for fmt and diff for toml files), for context taplo is used by the most popular extension in VScode [Even Better TOML](https://marketplace.visualstudio.com/items?itemName=tamasfe.even-better-toml
- Add contribution section to explain toml fmt with taplo.
 
Now to pass CI you need to run  `taplo fmt --option indent_string="    "` or if you use vscode have the `Even Better TOML` extension with 4 spaces for indent

### CODE REVIEW

File: tools/ci/src/main.rs [hunk 1]
```diff
@@ -73,6 +73,10 @@ fn main() {
         cmd!(sh, "cargo fmt --all -- --check")
             .run()
             .expect("Please run 'cargo fmt --all' to format your code.");
+
+        cmd!(sh, "taplo fmt --check --diff")
+            .run()
+            .expect("Please run 'taplo fmt'");
```
Comment (alice-i-cecile) [inline tools/ci/src/main.rs:79]: Mentioning the VSCode extension here is probably worthwhile.

**Comment by alice-i-cecile** [comment]
I would like to have `toml` format checking in CI, and this seems like a sensible tool and solution. It's annoying to deal with inconsistent formatting, as it either means hassling contributors to not auto-format or accepting churn.

**Comment by mockersf** [comment]
what would be the default formatting, without a config file?

**Comment by ameknite** [comment]
@alice-i-cecile It is already, or what do you mean?

**Comment by alice-i-cecile** [comment]
My previous comment was expressing my endorsement of this change :) I like the intent, but I am unsure about the details.

**Comment by ameknite** [comment]
@mockersf the indent is with 2 spaces, but 4 is more used in rust. I checked other repos like wgpu and winit, and both use 4 spaces

**Comment by mockersf** [comment]
I think I would prefer without a config file.

This should be mentioned in https://github.com/bevyengine/bevy/blob/main/CONTRIBUTING.md, with links to how to install and how to use it.

I'm not sure it should be plugged in our `ci` rust script, as it means it will start to fail for everyone who haven't installed that tool, even if they are not touching toml files. Maybe it should be like the markdown linter instead.

**Review by alice-i-cecile — APPROVED** [review]
I like this. IMO we should avoid duplicating this info in CONTRIBUTING.md, but if others want that, I don't feel strongly.

**Review by bushrat011899 — APPROVED** [review]
This is a great idea. I recently came across `taplo` when contributing to [`matchbox`](https://github.com/johanhelsing/matchbox/blob/b12e1099bb7c01a5d241982678d18ead8a84085c/.github/workflows/lint-toml.yml) and it was fantastic to have a standardised `Cargo.toml` file.

**Comment by alice-i-cecile** [comment]
@ameknite if you can resolve conflicts, I'm happy to merge this in :)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -163,6 +163,27 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install taplo
+        run: | 
+          curl -fsSL https://github.com/tamasfe/taplo/releases/latest/download/taplo-full-linux-x86_64.gz \
+          | gzip -d - \
+          | install -m 755 /dev/stdin /usr/local/bin/taplo
+      - name: Run Taplo
+        id: taplo
+        run: taplo fmt --check --diff --option indent_string="    "
```
Comment (mockersf) [inline .github/workflows/ci.yml:178]:
```suggestion
        run: taplo fmt --check --diff
```
I would prefer to stick to the defaults

Comment (mockersf) [inline .github/workflows/ci.yml:178]:
```suggestion
        run: taplo fmt --check --diff
```
and the other places where this is mentioned

**Review by mockersf — CHANGES_REQUESTED** [review]
It would be better to stick to the default settings

**Comment by daxpedda** [comment]
FWIW: custom settings could be specified in a [configuration file](https://taplo.tamasfe.dev/configuration/file.html) if that's desirable. The VSCode extensions also correctly detects and uses it.

**Comment by mockersf** [comment]
yup, but adding a new files on the root level adds clutter to the repository. IMO formatting is already nicer enough, aiming for the 4 spaces isn't worth either the clutter or the complexity

**Comment by rlidwka** [comment]
This is **not** a good change:

```diff
- bevy_reflect = { path = "../bevy_reflect", version = "0.12.0", features = ["bevy"] }
+ bevy_reflect = { path = "../bevy_reflect", version = "0.12.0", features = [
+    "bevy",
+ ] }
```

This is the first time I see features being split to a separate line (especially if there's only one feature), and it is less readable.

Overall, you are adding more tooling that every contributor has to install and run before making a pull request. It does not need to be standardized. It needs to be readable and easy to use, and to me it doesn't look like achieving either.

**Comment by rlidwka** [comment]
@ameknite, when I make a pull request, I grant access to that branch to bevy's maintainers ("Allowing edits by maintainers" checkbox on PR).

Can this thing be configured to format, commit and automatically push formatted files to that branch that's triggered CI (if it's pull-request and not master branch of course)?

**Comment by alice-i-cecile** [comment]
Yeah, we could add a bot of some form that either automatically does formatting, or does it on command. I'd be interested in exploring that for Rust/markdown/toml, as long as it's always in a separate commit.

**Comment by ameknite** [comment]
@mockersf 4 spaces is the standard in rust and it is uses in the majority of repos for toml files, even bevy, I don't know why you will consider it clutter, when is just the default settings for the majority of people.

If we change to 2 spaces, it will force to all the people to change to 2 spaces just to contribute to bevy. That is worse that just having a config file with the 4 spaces configuration.

Most people use vscode, for them this configuration does not affect them at all, just downloading an extension.

If we want to make it easier for people who use taplo with the command line, we should add the configuration in the command or even simpler by adding the configuration file to just run `taplo fmt`.

**Review by rlidwka — CHANGES_REQUESTED** [review]
```diff
- bevy_reflect = { path = "../bevy_reflect", version = "0.12.0", features = ["bevy"] }
+ bevy_reflect = { path = "../bevy_reflect", version = "0.12.0", features = [
+    "bevy",
+ ] }
```

I suggest to increase column_width (e.g. `--option column_width=90` or `100`) specifically to keep older one-line formatting of internal modules with `bevy` feature enabled.

This affects 20 changes in this PR (about half of it), and it's the only change that I can argue to be worse than before.

**Comment by DasLixou** [comment]
Coming from #10663 , could we add a Todo in the description to not forget to change the tracing doc link again?

**Comment by ameknite** [comment]
@rlidwka I would like to keep a simple fmt. It is not about achieving a beautiful format but rather  making it standardize and easy to use.

There reason why everyone agree to use cargo fmt is not because is pretty but because is standardize.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ameknite) [reply .github/workflows/ci.yml]: I change my mind, fixed.

**Comment by ameknite** [comment]
@DasLixou Fixed, thanks for mentioned it.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -175,13 +175,12 @@ jobs:
           | install -m 755 /dev/stdin /usr/local/bin/taplo
       - name: Run Taplo
         id: taplo
-        run: taplo fmt --check --diff --option indent_string="    "
```
Comment (bushrat011899) [inline .github/workflows/ci.yml]: I agree with this change. I think sticking with defaults unless we have a reason otherwise is a sensible choice.

**Review by bushrat011899 — APPROVED** [review]
Looks good to me!

**Review by mockersf — APPROVED** [review]

File: docs/profiling.md [hunk 4]
```diff
@@ -2,12 +2,14 @@
...
-  - [Perf flame graph](#perf-flame-graph)
-- [Compile time](#compile-time)
+- [Profiling](#profiling)
+  - [Table of Contents](#table-of-contents)
+  - [Runtime](#runtime)
+    - [Chrome tracing format](#chrome-tracing-format)
+    - [Tracy profiler](#tracy-profiler)
+    - [Adding your own spans](#adding-your-own-spans)
+    - [`perf` Flame Graph](#perf-flame-graph)
+  - [Compile time](#compile-time)
```
Comment (mockersf) [inline docs/profiling.md:12]: why those changes here?

Comment (alice-i-cecile) [inline docs/profiling.md:12]:
```suggestion
- [Runtime](#runtime)
  - [Chrome tracing format](#chrome-tracing-format)
  - [Tracy profiler](#tracy-profiler)
  - [Adding your own spans](#adding-your-own-spans)
  - [Perf flame graph](#perf-flame-graph)
- [Compile time](#compile-time)
```

**Comment by ameknite** [comment]
oh sorry, I didn't realize, it must have been the automatic formatting I have in vscode

**Pull request merged**